### DISCUSSION TITLE
Happy New Year and ZipChord 2.0

### DISCUSSION THREAD

**Discussion by psoukie** [psoukie/zipchord#68] [Announcements]
This is a short post to wish everyone all the best into 2023!

The upcoming version 2.0 of ZipChord will have two big features:

- Support for **"shorthands" autocomplete dictionaries** that expand regular typing. The current pre-release version comes with a dictionary based on 5,000 most frequent English words, allowing you to type e.g. "cpct" for "capacity".

- An **on-screen display with shortcut reminders** for existing chords and shorthands, so that when you manually type for example "when" it would unobtrusively remind you that you could have pressed `W`+`N` as a chord or typed a shorthand "whn" to save time. (If these were the shortcuts defined in your chord and shorthands dictionaries.) <ins>This is available as of 2.0 </ins>~~Alpha 4~~<ins>Beta!</ins>

I am also going to include ~~an automatic check and notification for when new versions of ZipChord are available,~~ <ins>a new UI for defining shortcuts, a more flexible dictionary format, and support for special keys in chords,</ins> and further improvements of the code. Special thanks to @NoobAndProud-01 for suggesting the auto-complete feature. 

The preview versions of ZipChord 2.0 are available from the [release page](https://github.com/psoukie/zipchord/releases).

**Comment by psoukie** [comment]
As of today, these -- and more -- features are now available in the [2.0 Beta release](https://github.com/psoukie/zipchord/releases/tag/2.0.0-beta).

For those using the previous stable versions, note that a few of the last versions of 1.9 published on GitHub had correctly attached .exe files, but they were pointing at and included the Alpha snapshots of 2.0. If this happened to you, get the correct latest [stable release 1.9.6](https://github.com/psoukie/zipchord/releases/tag/1.9.6).

**Comment by hiddengarden** [comment]
Happy new year to you too!

**Comment by psoukie** [comment]
And now the [Release Candidate](https://github.com/psoukie/zipchord/releases/tag/2.0.0-rc.2) is available.

**Comment by psoukie** [comment]
A quick update: The second release candidate was published yesterday, with fixes to all known issues in RC1.
